product_customer_code/views/product_customer_code_view.xml

```diff
@@ -0,0 +1,59 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo>
+
+    <record model="ir.ui.view" id="view_product_customer_code_form">
+        <field name="name">view.product.customer.code.form</field>
+        <field name="model">product.customer.code</field>
+        <field name="arch" type="xml">
+            <form string="Product Customer Code">
+                <field name="product_code"/>
+                <field name="partner_id"/>
+                <field name="product_id"/>
+                <field name="company_id" widget="selection" groups="base.group_multi_company"/>
+            </form>
+        </field>
+    </record>
+
+    <record model="ir.ui.view" id="view_product_customer_code_tree">
+        <field name="name">view.product.customer.code.tree</field>
+        <field name="model">product.customer.code</field>
+        <field name="arch" type="xml">
+            <tree string="Product Customer Code">
+                <field name="product_code"/>
+                <field name="partner_id"/>
+                <field name="product_id"/>
+                <field name="company_id"/>
+            </tree>
+        </field>
+    </record>
+
+    <record id="action_product_customer_code_tree" model="ir.actions.act_window">
+        <field name="name">Product Customer Code</field>
+        <field name="res_model">product.customer.code</field>
+        <field name="view_type">form</field>
+        <field name="view_id" ref="view_product_customer_code_tree"/>
+    </record>
+
+    <record id="view_product_customer_code_search" model="ir.ui.view">
+        <field name="name">view.product.customer.code.certificate.search</field>
+        <field name="model">product.customer.code</field>
+        <field name="arch" type="xml">
+            <search string="Product Customer Code">
+                <field name="product_code"/>
+                <field name="partner_id"/>
+                <field name="company_id"/>
+                <field name="product_id"/>
+                <newline/>
+                <group expand="0" string="Group By...">
+                    <filter string="Company" icon="terp-partner" domain="[]" context="{'group_by':'company'}"/>
+                    <filter string="Product" icon="terp-accessories-archiver" domain="[]" context="{'group_by':'product'}"/>
+                    <filter string="Partner" icon="terp-partner" domain="[]" context="{'group_by':'partner'}"/>
+                    <separator orientation="vertical"/>
+                </group>
+            </search>
+        </field>
+    </record>
+
+     <!-- <menuitem action="action_product_customer_code_tree" id="menu_product_customer_code" sequence="50" parent="product.menu_root" groups="group_product_customer_code_manager"/> -->
```

Comment:
Please remove the commented action then. We can redeclare the menu when needed, as is customary in the v10 (sale, purchase, inventory, etc. all define product menus)